Makefile

```diff
@@ -6,19 +6,41 @@ ifneq ($(OS),Windows_NT)
 	OS := $(shell uname)
 endif
 
+.DEFAULT_GOAL := help
+
+.ONESHELL:
```

Comment:
I'm only vaguely familiar with ONESHELL so I'm going to ask some potentially-dumb questions.

I know it means it'll run the whole definition in a single shell, so we need `-e` to exit early. But it doesn't affect how child targets behave, right? E.g. `make lint` will still fail on the first linter failure unless you use `-k`?

Reply:
There are no dumb questions!

Yep, you are correct.  `.ONESHELL` with `.SHELLFLAGS = -ec` behaves almost the same as standard make.  It gives you a slight performance improvement by not invoking a new shell for each line, but the main advantage is it lets you do multi-line shell commands (because all lines are invoked in the same shell).

Behavior for child targets is unaffected, and `-k` works the same as you'd expect it to.